relay-general/src/store/normalize.rs

```diff
@@ -3010,4 +3053,67 @@ mod tests {
         )
         "###);
     }
+
+    #[test]
+    fn test_keeps_valid_measurement() {
+        let name = "lcp";
+        let measurement = Measurement {
+            value: Annotated::new(420.69),
+            unit: Annotated::new(MetricUnit::Duration(DurationUnit::MilliSecond)),
+        };
+        let should_keep = true;
+
+        test_invalid_measurement(name, measurement, should_keep);
+    }
+
+    #[test]
+    fn test_drops_too_long_measurement_names() {
+        let name = "lcpppppppppppppppppppppppppppp";
+        let measurement = Measurement {
+            value: Annotated::new(420.69),
+            unit: Annotated::new(MetricUnit::Duration(DurationUnit::MilliSecond)),
+        };
+        let should_keep = false;
+
+        test_invalid_measurement(name, measurement, should_keep);
+    }
+
+    #[test]
+    fn test_drops_measurements_with_invalid_characters() {
+        let name = "i æm frøm nørwåy";
+        let measurement = Measurement {
+            value: Annotated::new(420.69),
+            unit: Annotated::new(MetricUnit::Duration(DurationUnit::MilliSecond)),
+        };
+        let should_keep = false;
+
+        test_invalid_measurement(name, measurement, should_keep);
+    }
+
+    fn test_invalid_measurement(name: &str, measurement: Measurement, should_keep: bool) {
+        let max_name_and_unit_len = Some(30);
+
+        let mut measurements: BTreeMap<String, Annotated<Measurement>> = Object::new();
+        measurements.insert(name.to_string(), Annotated::new(measurement));
+
+        let mut measurements = Measurements(measurements);
+        let mut meta = Meta::default();
+        let measurements_config = MeasurementsConfig {
+            max_custom_measurements: 1,
+            ..Default::default()
+        };
+
+        // Checks that there is 1 measurement before processing.
+        assert_eq!(measurements.len(), 1);
+
+        remove_invalid_measurements(
```

Comment:
Tests must not use part of the code that you're testing. Hardcoding the output in tests it's easy to understand and more solid than using the code.